tfhe/src/integer/client_key/mod.rs

```diff
@@ -425,13 +426,67 @@ impl ClientKey {
         self.key.encrypt(message)
     }
 
+    /// Encrypts a bool to a [BooleanBlock]
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use tfhe::integer::{gen_keys, BooleanBlock};
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// // We have 4 * 2 = 8 bits of message
+    /// let size = 4;
+    /// let (cks, sks) = gen_keys(PARAM_MESSAGE_2_CARRY_2_KS_PBS);
+    ///
+    /// let a = cks.encrypt_bool(false);
+    /// let dec = cks.decrypt_bool(&a);
+    /// assert_eq!(dec, false);
+    ///
+    /// let a = a.into_radix(size, &sks);
+    /// let dec: u64 = cks.decrypt_radix(&a);
+    /// assert_eq!(dec, 0);
+    /// ```
+    pub fn encrypt_bool(&self, msg: bool) -> BooleanBlock {
+        BooleanBlock::new_unchecked(self.encrypt_one_block(u64::from(msg)))
+    }
+
     /// Decrypts one block.
     ///
     /// This takes a shortint ciphertext as input.
     pub fn decrypt_one_block(&self, ct: &Ciphertext) -> u64 {
         self.key.decrypt(ct)
     }
 
+    /// Decrypts a ciphertext marked as holding a boolean value to a bool
+    ///
+    /// Treats 0 as false and the rest as true
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use tfhe::integer::{BooleanBlock, ClientKey};
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// let cks = ClientKey::new(PARAM_MESSAGE_2_CARRY_2_KS_PBS);
+    ///
+    /// let a = cks.encrypt_one_block(1u64);
+    /// let wrapped = BooleanBlock::new_unchecked(a);
+    /// let dec = cks.decrypt_bool(&wrapped);
+    /// assert_eq!(dec, true);
+    ///
+    /// // Here the value is not a boolean
+    /// // This is ok for decrypt function as it treats non zero as true
+    /// // However, generally when a function requires an input to be wrapped in
+    /// // BooleanBlock, it means the ciphertext must be 0 or 1
```

Comment:
Yes I was unsure about that, but if we don't show the thing, do we keep the `Treats 0 as false and the rest as true` or should it be a hidden detail since BooleanBlock is assumed to be 0 or 1 ?


 or maybe after decryption assert that the value is 0 or 1 before casting it to a bool ?

Reply:
Yeah I prefer having new_unchecked in case there's some cases where its the only choice (you don't want to pay for the price of convertion pbs).

Manual degree adjustment could be a potential solution, I mean, user can already do that to bypass the try_new, Its just 
that until noise and degree and not split it could be risky.

Reply:
ok so let's have a follow up issue so that degree gets fixed manually later ?

Reply:
> ok so let's have a follow up issue so that degree gets fixed manually later ?

You mean, where we remove new_unchecked from pub, and tel user to fix degree before calling try_new ?

Reply:
the first solution I would say

Reply:
alright status quo then thanks for the discussion